### DISCUSSION TITLE
What partition by to use for using an SnowflakeID / timestamp

### DISCUSSION THREAD

**Discussion by franz101** [ClickHouse/ClickHouse#64732] [Q&A]
I'm using the generateSnowflakeID which is pretty similiar to a timestamp

I could make the partition the the timestamp of it [:-7] which would be similiar to to YYYYMM

Or use a modulo. But would speed up look ups?

**Comment by den-crane** [comment]
ULID != SnowflakeID

**Reply by franz101** [reply]
changed title

**Comment by den-crane** [comment]
No difference.
I would use `Partition by (toYYYYMM(snowflakeToDateTime(id)))` for readability.

https://fiddle.clickhouse.com/b1408cf7-9853-4bc4-9b93-548ab074bf7c
